components/renku_data_services/session/models.py

```diff
@@ -17,10 +26,9 @@ class Member(BaseModel):
 
 
 @dataclass(frozen=True, eq=True, kw_only=True)
-class Environment(BaseModel):
-    """Session environment model."""
+class UnsavedEnvironment(BaseModel):
+    """Session environment model that isn't in the db yet."""
```

Comment:
suggestion: Since we are changing these, there is no reason whatsoever to use pydantic here. These can be just dataclasses. I already did this change in the PR that expands the environments and sessions.